woocommerce/woocommerce-ios
# REST API: Migrate product add-ons

Closes: #8668 


## Description

This PR migrates the product add-ons endpoint to use the REST API with application password authentication. The related mapper has also been updated to parse contents without the data envelope.

## Testing instructions

Pre-requisite: install [Product Add-on extension](https://href.li/?https://woocommerce.com/products/product-add-ons/) to your self-hosted store.
- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- On the prologue screen, select Enter your site address and proceed with the address of your test store.
- Log in with your site credentials. After the login succeeds, you should be navigated to the home screen.
- In Xcode console, you should not see any error log regarding product add-ons. If you use Charles Proxy or Proxyman, you should see that the request to `wc-product-add-ons/v1/product-add-ons` succeeds.

## Screenshots

No screenshot since the product endpoints haven't been migrated yet.

---
